In [2]:
import collections

name_counts = collections.Counter()
with open('./popular-names.txt', 'r') as f:
    for line in f:
        cols = line.strip().split('\t')
        if cols:
            name_counts[cols[0]] += 1

sorted_names = sorted(name_counts.items(), key=lambda item: (-item[1], item[0]))

for name, count in sorted_names:
    print(f"{count}\t{name}")

118	James
111	William
108	John
108	Robert
92	Mary
75	Charles
74	Michael
73	Elizabeth
70	Joseph
60	Margaret
58	George
58	Thomas
57	David
51	Richard
45	Helen
43	Christopher
43	Frank
41	Anna
40	Edward
39	Ruth
38	Patricia
37	Matthew
36	Dorothy
35	Emma
32	Barbara
31	Daniel
31	Joshua
26	Emily
26	Jennifer
26	Linda
26	Sarah
25	Jacob
25	Jessica
24	Betty
24	Mildred
24	Susan
23	Ashley
23	Henry
22	Nancy
21	Andrew
20	Amanda
20	Donald
20	Florence
20	Marie
19	Samantha
18	Karen
18	Lisa
18	Madison
18	Melissa
18	Olivia
17	Abigail
17	Stephanie
16	Ethel
16	Mark
16	Sandra
15	Angela
15	Carol
15	Ethan
15	Frances
15	Heather
15	Isabella
15	Michelle
14	Amy
14	Ava
14	Kimberly
14	Shirley
13	Brian
13	Deborah
13	Hannah
13	Jason
13	Nicole
13	Sophia
13	Virginia
12	Bertha
12	Donna
12	Minnie
11	Cynthia
10	Alice
10	Brittany
10	Doris
10	Mia
10	Nicholas
10	Noah
10	Ronald
9	Debra
9	Joan
9	Tyler
8	Alexander
8	Alexis
8	Clara
8	Ida
8	Judith
8	Mason
8	Taylor
7	Brandon
7	Harry
7	Liam
7	Sharon
7	Steven
7	Tammy
6	Anthony
5	Annie


In [ ]:
cut -f 1 hiro/chapter02//popular-names.txt | sort | uniq -c | sort -nr

 118 James
 111 William
 108 Robert
 108 John
  92 Mary
  75 Charles
  74 Michael
  73 Elizabeth
  70 Joseph
  60 Margaret
  58 Thomas
  58 George
  57 David
  51 Richard
  45 Helen
  43 Frank
  43 Christopher
  41 Anna
  40 Edward
  39 Ruth
  38 Patricia
  37 Matthew
  36 Dorothy
  35 Emma
  32 Barbara
  31 Joshua
  31 Daniel
  26 Sarah
  26 Linda
  26 Jennifer
  26 Emily
  25 Jessica
  25 Jacob
  24 Susan
  24 Mildred
  24 Betty
  23 Henry
  23 Ashley
  22 Nancy
  21 Andrew
  20 Marie
  20 Florence
  20 Donald
  20 Amanda
  19 Samantha
  18 Olivia
  18 Melissa
  18 Madison
  18 Lisa
  18 Karen
  17 Stephanie
  17 Abigail
  16 Sandra
  16 Mark
  16 Ethel
  15 Michelle
  15 Isabella
  15 Heather
  15 Frances
  15 Ethan
  15 Carol
  15 Angela
  14 Shirley
  14 Kimberly
  14 Ava
  14 Amy
  13 Virginia
  13 Sophia
  13 Nicole
  13 Jason
  13 Hannah
  13 Deborah
  13 Brian
  12 Minnie
  12 Donna
  12 Bertha
  11 Cynthia
  10 Ronald
  10 Noah
  10 Nicholas
  10 Mia
  10 Doris
  10 Brittany
  10 Alice
   9 Tyler
   9 Joan
   9 Debra
   8 Taylor
   8 Mason
   8 Judith
   8 Ida
   8 Clara
   8 Alexis
   8 Alexander
   7 Tammy
   7 Steven
   7 Sharon
   7 Liam
   7 Harry
   7 Brandon
   6 Anthony
   5 Jeffrey
   5 Jayden
   5 Gary
   5 Charlotte
   5 Annie
   4 Lillian
   4 Kathleen
   4 Justin
   4 Chloe
   4 Benjamin
   4 Austin
   3 Megan
   3 Harper
   3 Evelyn
   3 Elijah
   3 Aiden
   2 Rebecca
   2 Oliver
   2 Logan
   2 Lauren
   2 Larry
   2 Bessie
   2 Amelia
   1 Walter
   1 Tracy
   1 Scott
   1 Rachel
   1 Pamela
   1 Lucas
   1 Lori
   1 Laura
   1 Kelly
   1 Julie
   1 Crystal
   1 Carolyn